src/back/spv/writer.rs

```diff
@@ -1158,6 +1158,28 @@ impl Writer {
                             ),
                             left_lookup_ty,
                         ),
+                        (Dimension::Vector, Dimension::Vector)
```

Comment:
We can reduce code and have no DRY code, that is with using this match together with their scalar matching. So we would get:

```rust
(Dimension::Vector, Dimension::Vector) |
(Dimension::Scalar, Dimension::Scalar)
    if left_ty_inner.scalar_kind() == Some(crate::ScalarKind::Float) =>
{
    (
        super::instructions::instruction_f_mul(
            left_result_type_id,
            id,
            left_id,
            right_id,
        ),
        left_lookup_ty,
    )
}
(Dimension::Vector, Dimension::Vector) |
(Dimension::Scalar, Dimension::Scalar) => (
    super::instructions::instruction_i_mul(
        left_result_type_id,
        id,
        left_id,
        right_id,
    ),
    left_lookup_ty,
),
```